cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/chart_cache.go

```diff
@@ -139,11 +140,29 @@ func (c *ChartCache) SyncCharts(charts []models.Chart, clientOptions *common.Cli
 
 		// The tarball URL will always be the first URL in the repo.chartVersions.
 		// So says the helm plugin :-)
+		// however, not everybody agrees
+		// ref https://github.com/cjauvin/helm/blob/master/pkg/downloader/chart_downloader.go#L212
+		u, err := url.Parse(chart.ChartVersions[0].URLs[0])
+		if err != nil {
+			return fmt.Errorf("invalid URL format for chart [%s]: %v", chart.ID, err)
+		}
+
+		// If the URL is relative (no scheme), prepend the chart repo's base URL
+		// ref https://github.com/kubeapps/kubeapps/issues/4381
+		if !u.IsAbs() {
+			path := u.Path
+			u, err = url.Parse(chart.Repo.URL)
+			if err != nil {
+				return fmt.Errorf("invalid URL format for chart repo [%s]: %v", chart.ID, err)
+			}
+			u.Path = u.Path + path
+		}
```

Comment:
It's not clear to me if this is doing what you intend.? In particular, we shouldn't normally just join a url path via concatenation like that.

Actually, checking the link from the issue you've referenced, it looks like you've unintentionally add the change which *broke* helm (listed in the description of https://github.com/helm/helm/pull/3065 ), rather than the fix from 3065 itself? Is that possible?

Reply:
Well, I am looking at the latest helm code 
https://github.com/cjauvin/helm/blob/master/pkg/downloader/chart_downloader.go#L219
It is exactly as I have it

Reply:
> Well, I am looking at the latest helm code https://github.com/cjauvin/helm/blob/master/pkg/downloader/chart_downloader.go#L219 It is exactly as I have it

Gah never mind. The latest helm code does indeed look very different
https://github.com/helm/helm/blob/65d8e72504652e624948f74acbba71c51ac2e342/pkg/downloader/chart_downloader.go#L303
Let me fix this acocridngly

Reply:
Yep, I saw. Just didn't make sense to include the code that broke helm/helm#3065 rather than the code that fixed it. And I imagine the slash issue would hit us at some point if it affected helm users. Thanks.